### ISSUE TITLE
Hyperapp boilerplate using Fly and Rollup ✈️+🌯

### ISSUE THREAD

**Issue by tzellman** [hyperapp/hyperapp#55]
https://github.com/tzellman/hyperapp-boilerplate

I was inspired by @selfup to create a similar boilerplate / starting point, keeping the app structure almost intact.

I hope someone finds it useful.

One thing to point out: the app is still dependent on HyperApp 0.0.9, mainly for stability. I hope to update once things are more finalized with the latest changes.

**Comment by jbucaran** [comment]
@tzellman I can help out with the migration to HyperApp 0.0.12.

Are you planning to use JSX or Hyperx or both

**Comment by tzellman** [comment]
Up to this point I've only been dabbling with Hyperx, so likely that. I have been deeply rooted in Ember for the past 5 years so I'm not as up-to-date on competing technologies.

But, yes, any help or interest would be greatly appreciated!

**Comment by tzellman** [comment]
I made the [changes](https://github.com/tzellman/hyperapp-boilerplate/commit/e8f81d04f866a0143413c75a81d42ec1ae178657) to update to hyperapp 0.0.12. Let me know if there is a more preferred way.

**Comment by jbucaran** [comment]
LGTM:)

**Comment by selfup** [comment]
@tzellman This is great! 🎉

**Comment by jbucaran** [comment]
@tzellman I finally came up with a rollup+hyperx+hyperxify build setup that works, perhaps you can update your boilerplate?

* https://gist.github.com/jbucaran/fac2c3de24e5171596fb189f9c1feb8e

The setup also uses hyperxify to transform hyperx into native `h` calls, which is a must have.

The only downside is, despite using rollup, we can't do treeshaking to hyperapp because in order to use hyperx, I had also to use commonjs.

tl;dr, if anyone is using rollup with hyperapp, the ideal combo would be hyperapp + jsx. For hyperx the best will always be browserify (although webpack is also okay).

* webpack+hyperx: https://gist.github.com/jbucaran/c6a6bdb5383a985cec6b0ae4ebe5a4b1
* browserify+hyperx: https://gist.github.com/jbucaran/48c1edb4fb0ea1aa5415b6686cc7fb45

**Comment by tzellman** [comment]
Cool, I will look into doing that! I have no allegiance to hyperx, so I may just update to use JSX. Hopefully will have some time later... :hourglass_flowing_sand:

**Comment by tzellman** [comment]
So, dumb question - how does one use hyperxify (with the browserify rollup plugin) when the main (entry js) file uses ES6 `import` statements instead of `require`?

My assumption is that I would first need to transpile (using buble or babel), however I still am getting an error during the browserify/hyperxify step.

Your gist uses require. I pulled down the zip and modified it to use ES6 imports, and am having the same problem. Maybe you can lend me some insight?

**Comment by jbucaran** [comment]
@tzellman Exactly! I also wanted to do this, so rollup could do tree-shaking (e.g. get rid of the router when not using it), but hyperx is commonjs and needs to be used in the same file where you import hyperapp. 

I also tried mixing both import and require in the same file, but nope.

In conclusion, I think it's not possible. Which is why I found rollup+hyperx combination to be the least desirable.

* For JSX: browserify, webpack or rollup is perfect
* For Hyperx: webpack or browserify are best.

**Comment by tzellman** [comment]
Yeah, so converting to JSX seemed to be the most sane solution; also created a much smaller final payload.

I was able to keep all of my existing tooling: fly driving rollup with buble.

Thanks for all of the useful info!

**Comment by tunnckoCore** [comment]
Not aggressive tone, but why the hell will use rollup with browserify plugin :D it just not make sense.

But yea, it seems that JSX is better in many cases. For bad or for good - it is fact and maybe the main reason of its explosion and using it everywhere, another reason for sure is that plain html is mega easily convertable to jsx.

**Comment by tzellman** [comment]
For the past 5 years I've only developed rich front-ends with Ember, so I am learning some of the other tooling and libraries. I was drawn to the simplicity of Hyperapp as well as the positive attitude of the maintainer @jbucaran - so I am trying to learn from others here.

And I am not using browserify. I attempted to use it just so I could handle the hyperxify transform as a pre-processing step. It wasn't worth the hassle.

**Comment by jbucaran** [comment]
@tunnckoCore I think this comment ([link](https://github.com/hyperapp/hyperapp/issues/55#issuecomment-281684297)) was unnecessary and adds nothing to the conversation.

 >Not aggressive tone, but why the hell will use rollup with browserify plugin :D it just not make sense.
>
>But yea, it seems that JSX is better in many cases. For bad or for good - it is fact and maybe the main reason of its explosion and using it everywhere, another reason for sure is that plain html is mega easily convertable to jsx.

This thread's purpose is to inform the community about @tzellman's HyperApp boilerplate, not start a discussion about JSX vs Hyperx. 

Let's all be civil and remember that issues are part of this project's documentation. If you want to discuss JSX vs Hyperx, you can do it, but somewhere else. For example, HyperApp's Slack room's #random channel, your blog, etc.

**Comment by tunnckoCore** [comment]
As i said, no bad feelings and don't get it aggressive. I was just curious and kinda clarification. I'm not JSX nor HyperX fan, but i kinda see why JSX is so used, didn't get it until now (releasing of hyperapp)

So cheers. :)
